etna/datasets/tsdataset.py

```diff
@@ -1111,21 +1113,29 @@ def get_level_dataset(self, target_level: str) -> "TSDataset":
         if target_level_index > current_level_index:
             raise ValueError("Target level should be higher in the hierarchy than the current level of dataframe!")
 
+        target_names = tuple(get_target_with_quantiles(columns=self.columns))
+        if self.target_components_names is not None:
+            target_names += tuple(set(self.target_components_names))
+
         if target_level_index < current_level_index:
             summing_matrix = self.hierarchical_structure.get_summing_matrix(
                 target_level=target_level, source_level=self.current_df_level
             )
 
             target_level_df = get_level_dataframe(
-                df=self.df,
+                df=self.to_pandas(features=target_names),
                 mapping_matrix=summing_matrix,
                 source_level_segments=current_level_segments,
                 target_level_segments=target_level_segments,
             )
 
         else:
-            target_names = tuple(get_target_with_quantiles(columns=self.columns))
-            target_level_df = self[:, current_level_segments, target_names]
+            target_level_df = self.to_pandas(features=target_names)
+
+        target_components_df = None
+        if self.target_components_names is not None:
```

Comment:
There are a lot of this check: `if self.target_components_names is not None:`, can we somehow avoid it?

Reply:
Not sure that is it possible as all the logic can't be placed inside one if

Reply:
Change default value of `target_components_names`